frontend/express/public/javascripts/countly/countly.common.js

```diff
@@ -205,6 +205,31 @@
             });
         };
 
+        /**
+         * Adds notification toast to the list.
+         * @param {*} payload notification toast
+         *  payload.color: color of the notification toast
+         *  payload.text: text of the notification toast
+         */
+        countlyCommon.dispatchNotificationToast = function(payload) {
+            if (window.countlyVue && window.countlyVue.vuex) {
+                var currentStore = window.countlyVue.vuex.getGlobalStore();
+                if (currentStore) {
+                    currentStore.dispatch('countlyCommon/onAddNotificationToast', payload);
+                }
+            }
+        };
+
+        /**
+         * Generates unique id string using unsigned integer array.
+         * @returns {string} unique id
+         */
+        countlyCommon.generateId = function() {
+            var crypto = window.crypto || window.msCrypto;
+            var uint32 = crypto.getRandomValues(new Uint32Array(1))[0];
```

Comment:
still good enough, but we could simply use a counter too :slightly_smiling_face:

Reply:
that's what I initially had. My thinking was that I wanted to make the ids more robust while also not polluting the common vuex module with counter state